ktor-server/ktor-server-plugins/ktor-server-di/common/src/io/ktor/server/plugins/di/DependencyProvider.kt

```
@@ -205,6 +205,7 @@ public open class MapDependencyProvider(
 
     override fun <T> set(key: DependencyKey, value: DependencyResolver.() -> T) {
         val create = ExplicitCreateFunction(key, value as DependencyResolver.() -> Any)
+        log.debug { "Provided $key ${RuntimeException().externalTraceLine()}" }
```

detekt (1.23.7)

[warning] A call to the default constructor of an exception was detected. Instead one of the constructor overloads should be called. This allows to provide more meaningful exceptions.

(detekt.exceptions.ThrowingExceptionsWithoutMessageOrCause)